src/orderly/read.py

```diff
@@ -0,0 +1,65 @@
+import ast
+
+
+def orderly_read(path):
+    with open(path) as f:
+        src = ast.parse(f.read())
+    return _read_py(src)
+
+
+# In the R version of this function we do a more involved read, trying
+# to handle "static" versions of all of the orderly core
+# functions. However,we then never actually used that for anything, so
+# to avoid overcomplication we'll do the least possible here which is
+# just to read the parameters from the file, verify that it is only
+# called once, and that is called at the top-level.
+#
+# The return value does nod at future extension though.
+def _read_py(src):
+    ret = {"parameters": []}
+    for expr in src.body:
+        dat = _read_expr(expr)
+        if dat:
+            if dat["name"] == "parameters" and ret["parameters"]:
+                msg = f"Duplicate call to 'parameters()' on line {expr.lineno}"
+                raise Exception(msg)
+            ret["parameters"].append(dat["data"])
```

Comment:
This line should be inside the `if dat["name"] == "parameters"` condition I think.